### PR TITLE
Refactored common IO to math-scala

### PR DESCRIPTION
**Pull request by smarthi**

### CODE REVIEW

**Comment by andrewpalumbo** [comment]
Hey Suneel,

Tested some stuff out in the shell (on Spark 1.3.1 in single machine standalone mode) and it seems to be working without issue:
 
```
mahout> val a= Matrices.uniformView(30,4,1)
mahout> val b = Matrices.gaussianView(1,4,1)
mahout> val drmB = drmParallelize(b, numPartitions = 2 )
mahout> val drmA=drmParallelize(a, numPartitions = 2)
mahout> val drmC = drmA %*% drmB.t
mahout> drmC.collect
res6: org.apache.mahout.math.Matrix =                                           
{
 0 =>	{0:1.5915372524919211}
 1 =>	{0:2.129830984242081}
 2 =>	{0:1.4224061257679534}
 3 =>	{0:1.3390522838114067}
 4 =>	{0:1.071621716858415}
 5 =>	{0:1.6662158252690769}
 6 =>	{0:1.3615709513317833}
 7 =>	{0:1.1913478272651996}
 8 =>	{0:1.3437170178001066}
 9 =>	{0:1.2379076792678294}
... }


mahout> val tfidfDRM = drmDfsRead("/tmp/mahout-work-wiki/wikipediaVecs/tfidf-vectors/")
tfidfDRM: org.apache.mahout.math.drm.CheckpointedDrm[_] = org.apache.mahout.sparkbindings.drm.CheckpointedDrmSpark@13bbe068
mahout> tfidfDRM.colSums
res1: org.apache.mahout.math.Vector = {1:57.075862884521484,2:71.80289077758789,3:18.562942504882812,4:13.699396133422852,5:18.562942504882812,6:13.699396133422852,7:13.699396133422852,8:13.699396133422852,9:13.699396133422852,10:13.699396133422852,11:13.699396133422852,12:13.699396133422852,13:13.699396133422852,14:13.699396133422852,15:57.114768981933594,16:38.715500831604004,17:13.699396133422852,18:30.035472869873047,19:13.699396133422852,20:26.251964569091797,21:78.23203754425049,22:13.699396133422852,23:13.699396133422852,24:13.699396133422852,25:13.699396133422852,26:13.699396133422852,27:50.00976371765137,28:13.699396133422852,29:13.699396133422852,30:13.699396133422852,31:13.699396133422852,32:13.699396133422852,33:13.699396133422852,34:13.699396133422852,35:13.699396133422852,...
```

ran mahout spark-trainnb and mahout-testnb both seemed fine.

**Comment by smarthi** [comment]
Thanks @andrewpalumbo, so its safe to merge this PR into Flink branch then.

File: spark/src/test/scala/org/apache/mahout/sparkbindings/io/IOSuite.scala [hunk 1]
```diff
@@ -187,7 +187,7 @@ object IOSuite {
     val input = new Input(new ByteArrayInputStream(baos.toByteArray))
 
     def outStream: Stream[T] =
-      if (input.eof) Stream.empty
+      if (input.available() == 0) Stream.empty
```
Comment (dlyubimov) [inline spark/src/test/scala/org/apache/mahout/sparkbindings/io/IOSuite.scala]: You sure? I never quite got those kryo streams. in regular streams, available != eof

**Comment by dlyubimov** [comment]
General: +1 (salute to the newly born Scala style cop :) :+1:

File: spark/src/test/scala/org/apache/mahout/sparkbindings/io/IOSuite.scala (hunk 1 above)
Comment (smarthi) [inline spark/src/test/scala/org/apache/mahout/sparkbindings/io/IOSuite.scala]: eof is missing in Kryo 2 apparently, couldn't quite figure out what went there?

**Pull request closed without merging**